src/main/com/fulcrologic/fulcro/algorithms/normalized_state_helpers.cljc

```diff
@@ -0,0 +1,441 @@
+      (dissoc-in state-map ident)
+      idents))
+
+
+
+  (comment
+
+
+
+
+  (let [item (get-in state-map ident)
+        idents (extract-all-idents item)]
+
+    (reduce
+      (fn [s i] (deep-remove-entity* s i))
+      (dissoc-in state-map ident)
+      idents))
+
+
+  (let [path [:person/id 1]]
+    (cond-> normalized-state
+     (clojure.core/get-in normalized-state (butlast path))
+     (update-in (butlast path) dissoc (second path))))
+
+
+
+  (let [path [:person/id 1]]
+    (cond-> normalized-state
+            (clojure.core/get-in normalized-state (butlast path))
+            (update-in normalized-state (butlast path) dissoc (second path)))))
+
+  (let [path [:person/id 1 :person/spouse]]
+    (clojure.core/get-in normalized-state (butlast path))
+    (update-in normalized-state (butlast path) dissoc (second path)))
+
+
+  (update-in normalized-state (butlast [:person/id 1 :person/spouse]) dissoc)
+
+  (update-in normalized-state (butlast [:person/id 1 :person/spouse]) dissoc 1)
+
+
+  (dissoc normalized-state :person/id)
+
+  (dissoc-in normalized-state [:person/id 1])
+
+  (update-in normalized-state [:person/id] dissoc 1)
+
+  (update-in normalized-state [:person/id] dissoc 1 2)
+
+  (sequence cat (repeat ["tom" "nick" "jane"]))
+
+
+
+  (extract-all-idents (get-in normalized-state [:person/id 1]))
+
+  (vals (get-in normalized-state [:person/id 1]))
+
+  (deep-remove-entity* normalized-state [:person/id 1])
+
+  (remove-entity* normalized-state [:person/id 1])
+
+  '())
+
+
+
+;;============================================================================
+
+;; TODO
+;; - [ ] `sort-identy-by`
+;;    (sort-idents-by :entity/field vector-of-idents)
+;;    This can enable
+;;    (swap! state update-in [:entity 1 :list] sort-idents-by :list/field)
+
+
+
+;;============================================================================
+
+;; TODO path-to-edge...should that be tree-path or db-path?
+;; NOTE This is pretty much remove-ident*, but with the option of recursively removing the stuff
+
+;; NOTE pointed to...might need a better name.
+(defn remove-edge*
+  ;; TODO
+  ([state-map path-to-edge])
+
+  ;; TODO needs cascade argument like remove-entity*
+  ([state-map path-to-edge cascade])
+  )
+
+(comment
+
+  (defn remove-ident*
+    "Removes an ident, if it exists, from a list of idents in app state. This
+    function is safe to use within mutations."
+    [state-map ident path-to-idents]
+    {:pre [(map? state-map)]}
+    (let [new-list (fn [old-list]
+                     (vec (filter #(not= ident %) old-list)))]
+      (update-in state-map path-to-idents new-list)))
+
+  )
+
+
+
+;============================================================================
+;; REMOVE
+(defsc A [this props]
+  {:ident         :person/id
+   :query         [:person/id :person/name]
+   :initial-state {:person/id 1 :person/name "Alice"}
+   :extra-data    42}
+  (dom/div "TODO"))
+
+
+(def ui-a (comp/factory A))
+
+
+(defn ui->props
```

Comment:
Make a very simple component, possibly with a little nesting. Make a very simple normalized db that has data that would work for it, and then show it can pull the props for you.

Don't test the side that takes `this`